File: ch08/App_Code/CS/Listing8-56.cs
using System;
using System.CodeDom;
using System.Web.Compilation;
using System.Web.UI;

namespace Wrox.CS
{
[ExpressionPrefix("MyFirstCustomExpression")]
[ExpressionEditor("MyFirstCustomExpressionEditor")]
public class MyFirstCustomExpression : ExpressionBuilder
{
    public override System.CodeDom.CodeExpression
        GetCodeExpression(BoundPropertyEntry entry, object parsedData,
            ExpressionBuilderContext context)
    {
        return new CodeCastExpression("Int64", 
            new CodePrimitiveExpression(1000));
    }
}
}

File: ch08/App_Code/VB/Listing8-56.vb
Imports System
Imports System.CodeDom
Imports System.Web.Compilation
Imports System.Web.UI

Namespace Wrox.VB

    <ExpressionPrefix("MyCustomExpression")>
    <ExpressionEditor("MyCustomExpressionEditor")>
    Public Class MyFirstCustomExpression
        Inherits ExpressionBuilder

        Public Overrides Function GetCodeExpression(ByVal entry As BoundPropertyEntry,
            ByVal parsedData As Object, ByVal context As ExpressionBuilderContext) _
            As System.CodeDom.CodeExpression

            Return New CodeCastExpression("Int64", New CodePrimitiveExpression(1000))
        End Function
    End Class

End Namespace
